src/components/TextArea/index.js

```diff
@@ -0,0 +1,67 @@
+import { Component } from 'pet-dex-utilities';
+import './index.scss';
+
+const events = [
+  'setName',
+  'setPlaceholder',
+  'setMaxLength',
+  'setRequired',
+  'error',
+];
+
+const html = `
+  <div class="textarea-container">
+    <textarea class="textarea-container__textarea" data-select="textarea" rows="1" cols="1"></textarea>
+    <span class="textarea-container__resize-trigger" data-select="resize-trigger" contenteditable></span>
+  </div>
+`;
+
+export default function TextArea({ name, placeholder, maxLength, required }) {
+  Component.call(this, { html, events });
+  const $textarea = this.selected.get('textarea');
+
+  this.setName(name);
+  this.setPlaceholder(placeholder);
+  this.setMaxLength(maxLength);
+  this.setRequired(required);
+
+  $textarea.addEventListener('focus', () =>
+    $textarea.classList.remove('error'),
+  );
+  $textarea.addEventListener('input', () => this.autoResize());
+  window.addEventListener('resize', () => this.autoResize());
+}
+
+TextArea.prototype = Object.assign(TextArea, Component.prototype, {
+  setName(name = '') {
+    this.selected.get('textarea').name = name;
+    this.emit('setName', name);
+  },
+  setPlaceholder(placeholder = '') {
+    this.selected.get('textarea').placeholder = placeholder;
+    this.emit('setPlaceholder', placeholder);
+  },
+  setMaxLength(maxLength) {
+    if (maxLength) {
+      this.selected.get('textarea').maxLength = maxLength;
+    }
+    this.emit('setMaxLength', maxLength);
+  },
+  setRequired(required = false) {
+    this.selected.get('textarea').required = required;
+    this.emit('setRequired', required);
+  },
+  error() {
+    this.selected.get('textarea').classList.add('error');
+    this.emit('error');
+  },
+  autoResize() {

```

Comment:
Não acho que esse tipo de método deva ser publico.
Se for deixar público renomeie para onResize() que é o padrão para callback que estamos usando